store/sqlstore/file_info_store.go

```diff
@@ -614,22 +625,38 @@ func (fs SqlFileInfoStore) Search(paramsList []*model.SearchParams, userId, team
 		if terms == "" && excludedTerms == "" {
 			// we've already confirmed that we have a channel or user to search for
 		} else if fs.DriverName() == model.DatabaseDriverPostgres {
-			// Parse text for wildcards
-			if wildcard, err := regexp.Compile(`\*($| )`); err == nil {
-				terms = wildcard.ReplaceAllLiteralString(terms, ":* ")
-				excludedTerms = wildcard.ReplaceAllLiteralString(excludedTerms, ":* ")
-			}
+			// replace a term only if there is a wildcard at the end of the word
+			excludedTerms = wildcardRegExpForPostgresExcludedTerms.ReplaceAllLiteralString(excludedTerms, ":* ")
 
 			excludeClause := ""
 			if excludedTerms != "" {
 				excludeClause = " & !(" + strings.Join(strings.Fields(excludedTerms), " | ") + ")"
 			}
 
+			exactPhraseTerms := []string{}
+
+			// Since we will put wildcard on the rest of the terms,
+			// we will get exactPhraseTerms first and then remove the matching terms
+			for _, term := range exactPhraseRegExpForFileSearch.FindAllString(terms, -1) {
+				quoteRemovedTerm := strings.Trim(term, "\"")
+				exactPhraseTerms = append(exactPhraseTerms, strings.Fields(quoteRemovedTerm)...)
+			}
+
+			terms = exactPhraseRegExpForFileSearch.ReplaceAllLiteralString(terms, "")
+
+			wildcardAddedTerms := strings.Fields(terms)
+
+			for index, term := range wildcardAddedTerms {
+				// A wildcard is attached to the end of every word
+				// regardless of whether or not a wildcard exists
+				wildcardAddedTerms[index] = wildcardRegExpForFileSearch.ReplaceAllLiteralString(term, ":*")
+			}
```

Comment:
Similarly: is this the same as
```suggestion
			for index, term := range wildcardAddedTerms {
				// A wildcard is attached to the end of every word
				// regardless of whether or not a wildcard exists
				if !strings.HasSuffix(term, "*") {
					wildcardAddedTerms[index] = term + ":*"
				}
			}
```